src/Components/Blazor/Blazor/test/WebAssemblyUriHelperTest.cs

```diff
@@ -29,6 +29,9 @@ public void ComputesCorrectBaseUri(string baseUri, string expectedResult)
         [InlineData("scheme://host/path/", "scheme://host/path/", "")]
         [InlineData("scheme://host/path/", "scheme://host/path/more", "more")]
         [InlineData("scheme://host/path/", "scheme://host/path", "")]
+        [InlineData("scheme://host/path/", "scheme://host/path#hash", "#hash")]
+        [InlineData("scheme://host/path/", "scheme://host/path/#hash", "#hash")]
+        [InlineData("scheme://host/path/", "scheme://host/path/more#hash", "more#hash")]
```

Comment:
Support for these cases was needed for the E2E tests. It's because we use a `#server` suffix to trigger server-side execution (and always have done - that's not new), and it turns out the `ToBaseRelativePath` logic didn't understand the case that's now on line 32 here.